### ISSUE TITLE
Examples checking scripts report false errors for certain properties

### ISSUE THREAD

**Issue by RichardWallis** [schemaorg/schemaorg#2191]
Identified in the process to reduce the number of examples errors reported by Travis CI scripts (see #2189), certain [Schema.org] valid property assignments are reported as being errors.

Specifically:

- URL:  Errors are reported if value assigned, to a property expecting a URL, is a relative URL (one that does not begin 'http://' or 'https://')

- Duration: Errors are reported when a string is assigned, correctly (ISO8601) or not, as a value for a property expecting type Duration.

- DateTime: Errors are reported for valid assignment of DateTime values eg. ```Object "2016-06-15T00:01" not compatible with rangeIncludes (schema:DateTime)```

- Boolean: Errors falsely reported for boolean values eg. (sdo-tv-listing-examples.txt[184] - tv-listing4 (microdata)): ```Object "http://schema.org/True" not compatible with rangeIncludes (schema:Boolean)```

/cc @gkellogg

**Comment by gkellogg** [comment]
Can you provide me with specific failing tests for those above? We shouldn’t see relative URIs, as everything should be resolved against the file location.

**Comment by RichardWallis** [comment]
@gkellogg  URL example:
```
 39) Examples sdo-course-examples.txt[503] - Course-courseCode-provider-107 (jsonld)
      # lint error
      Failure/Error: specify("sdo-course-examples.txt[503] - Course-courseCode-provider-107 (jsonld)") {expect("/home/travis/build/schemaorg/schemaorg/scripts/spec/data/Course-courseCode-provider-107-jsonld.html").to lint_cleanly}
      
        Source: <script type="application/ld+json">
        {
          "@context": "http://schema.org/",
          "@type": "schema:Course",
          "schema:courseCode": "F300",
          "schema:name": "Physics",
          "schema:provider": {
            "@type": "schema:CollegeOrUniversity",
            "schema:name": "University of Bristol",
            "schema:url": "provider/324/university-of-bristol"
          }
        }
        </script>
      
      
        Results:
        property  schema:url
          Object "provider/324/university-of-bristol" not compatible with rangeIncludes (schema:URL)
```

And:
```
 41) Examples examples.txt[2130] - MusicPlaylist-131 (rdfa)
      # lint error
      Failure/Error: specify("examples.txt[2130] - MusicPlaylist-131 (rdfa)") {expect("/home/travis/build/schemaorg/schemaorg/scripts/spec/data/MusicPlaylist-131-rdfa.html").to lint_cleanly}
      
        Source: <div vocab="http://schema.org/" typeof="MusicPlaylist">
          <span property="name">Classic Rock Playlist</span>
          <meta property="numTracks" content="5"/>
...     
          <div property="track" typeof="MusicRecording">
            3.<span property="name">Sharp Dressed Man</span> -
            <span property="byArtist">ZZ Top</span>
            <meta content="sharp-dressed-man" property="url" />
            <meta content="PT4M13S" property="duration" />
            <meta content="Eliminator" property="inAlbum" />
         </div>
 ...
      <div property="track" typeof="MusicRecording">
            5.<span property="name">Hurt So Good</span> -
            <span property="byArtist">John Cougar</span>
            <meta content="hurt-so-good" property="url" />
            <meta content="PT3M39S" property="duration" />
            <meta content="American Fool" property="inAlbum" />
          </div>
        </div>
      
      
        Results:
        property  schema:url
          Object "sharp-dressed-man" not compatible with rangeIncludes (schema:URL)
...
          Object "hurt-so-good" not compatible with rangeIncludes (schema:URL)
      # ./spec/examples_spec.rb:461:in `block (2 levels) in <top (required)>'
```

**Comment by gkellogg** [comment]
On Apr 6, 2019, at 12:30 AM, Richard Wallis <notifications@github.com> wrote:
> 
> @gkellogg URL example:
> 
>  39) Examples sdo-course-examples.txt[503] - Course-courseCode-provider-107 (jsonld)
>       # lint error
>       Failure/Error: specify("sdo-course-examples.txt[503] - Course-courseCode-provider-107 (jsonld)") {expect("/home/travis/build/schemaorg/schemaorg/scripts/spec/data/Course-courseCode-provider-107-jsonld.html").to lint_cleanly}
>       
>         Source: <script type="application/ld+json">
>         {
>           "@context": "http://schema.org/",
>           "@type": "schema:Course",
>           "schema:courseCode": "F300",
>           "schema:name": "Physics",
>           "schema:provider": {
>             "@type": "schema:CollegeOrUniversity",
>             "schema:name": "University of Bristol",
>             "schema:url": "provider/324/university-of-bristol"
>           }
>         }
>         </script>
>       
>       
>         Results:
>         property  schema:url
>           Object "provider/324/university-of-bristol" not compatible with rangeIncludes (schema:URL)
This is because schema:url is not a term with @type: @id, so it is interpreted as text, while the range of schema:url is URL. Either use an expanded value for the relative IRI, or use simply “url”, which is a term.
> And:
>  41) Examples examples.txt[2130] - MusicPlaylist-131 (rdfa)
>       # lint error
>       Failure/Error: specify("examples.txt[2130] - MusicPlaylist-131 (rdfa)") {expect("/home/travis/build/schemaorg/schemaorg/scripts/spec/data/MusicPlaylist-131-rdfa.html").to lint_cleanly}
>       
>         Source: <div vocab="http://schema.org/" typeof="MusicPlaylist">
>           <span property="name">Classic Rock Playlist</span>
>           <meta property="numTracks" content="5"/>
> ...     
>           <div property="track" typeof="MusicRecording">
>             3.<span property="name">Sharp Dressed Man</span> -
>             <span property="byArtist">ZZ Top</span>
>             <meta content="sharp-dressed-man" property="url" />
>             <meta content="PT4M13S" property="duration" />
>             <meta content="Eliminator" property="inAlbum" />
>          </div>
>  ...
>       <div property="track" typeof="MusicRecording">
>             5.<span property="name">Hurt So Good</span> -
>             <span property="byArtist">John Cougar</span>
>             <meta content="hurt-so-good" property="url" />
>             <meta content="PT3M39S" property="duration" />
>             <meta content="American Fool" property="inAlbum" />
>           </div>
>         </div>
>       
>       
>         Results:
>         property  schema:url
>           Object "sharp-dressed-man" not compatible with rangeIncludes (schema:URL)
> ...
>           Object "hurt-so-good" not compatible with rangeIncludes (schema:URL)
>       # ./spec/examples_spec.rb:461:in `block (2 levels) in <top (required)>'

Missing @datatype: schema:URL,  believe.

Gregg

**Comment by gkellogg** [comment]
@RichardWallis do we still think there are cases where the Ruby example checking scripts are getting it wrong?

**Comment by RichardWallis** [comment]
Post V3.6 I am going to dive into the Travis logs again (a man can only take so many lint reports at a time!) to see what we can do to get the numbers down.

I'll start with a fresh eye to see where it leads me.
